client/my-sites/stats/summary/index.jsx

```diff
@@ -43,6 +47,28 @@ const StatsStrings = statsStringsFactory();
 class StatsSummary extends Component {
 	componentDidMount() {
 		window.scrollTo( 0, 0 );
+
+		const { context, period } = this.props;
+		const { module } = context.params;
+		const { query } = context;
+
+		recordCurrentScreen( module, {
+			queryParams: query,
+			period: period?.period,
+		} );
+	}
+
+	componentDidUpdate( prevProps ) {
+		if ( ! isEqual( prevProps.statsQueryOptions, this.props.statsQueryOptions ) ) {
```

Comment:
```suggestion
		if ( ! isEqual( prevProps.context.query, this.props.context.query ) ) {
```
Should we just compare `context.query` because they will be recorded?